Sanitizer error in ydb/tests/functional/api/test_discovery.py.TestDiscoveryExtEndpoint.test_scenario

ydb/tests/functional/api/test_discovery.py.TestDiscoveryExtEndpoint.test_scenario

**Build Information:**
- Version: 16ccbe8aee73119b6b3bf5af118feba55ba69799
- Build Preset: release-tsan
- Branch: prestable-25-3

**Run Command:**
```
./ya make -ttt --build "release" --sanitize="thread" -DDEBUGINFO_LINES_ONLY -F 'test_discovery.py::TestDiscoveryExtEndpoint::test_scenario' ydb/tests/functional/api
```

**Owner:** [TEAM:@ydb-platform/](https://github.com/orgs/ydb-platform/teams/)

**Test run history:** [link](https://datalens.yandex/34xnbsom67hcq?branch=prestable-25-3&full_name=ydb/tests/functional/api/test_discovery.py.TestDiscoveryExtEndpoint.test_scenario&build_type=release-tsan)

More info in [dashboard](https://datalens.yandex/4un3zdm0zcnyr)

**Logs:**
- [log](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_Small_and_Medium/17708525640/ya-prestable-25-3-x86-64-tsan/try_3/artifacts/logs/ydb/tests/functional/api/test-results/py3test/testing_out_stuff/test_discovery.py.TestDiscoveryExtEndpoint.test_scenario.log)
- [logsdir](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_Small_and_Medium/17708525640/ya-prestable-25-3-x86-64-tsan/try_3/test_artifacts/ydb-tests-functional-api.zip)

**Error Summary:**
```
teardown failed:
ydb/tests/functional/api/test_discovery.py:44: in teardown_class
    cls.cluster.stop()
ydb/tests/library/harness/kikimr_runner.py:639: in stop
    raise daemon.SeveralDaemonErrors(saved_exceptions)
E   ydb.tests.library.harness.daemon.SeveralDaemonErrors: Daemon failed with message: Bad exit_code..
E   Process exit_code = 100.
E   Stdout file name: 
E   /home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/functional/api/test-results/py3test/testing_out_stuff/test_discovery.py.TestDiscoveryExtEndpoint.test_scenario/cluster/slot_2/stdout
E   Stderr file name: 
E   /home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/functional/api/test-results/py3test/testing_out_stuff/test_discovery.py.TestDiscoveryExtEndpoint.test_scenario/cluster/slot_2/stderr
E   Stderr content:
E   
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   Current KQP shutdown state: spent 0 seconds, not started yet
E   warning: address range table at offset 0x880 has a premature terminator entry at offset 0x890
E   ==================
E   WARNING: ThreadSanitizer: data race (pid=1431254)
E   Read of size 8 at 0x724400044cb8 by thread T156:
E   #0 NActors::TMon::RegisterActorHandler(NActors::TMon::TRegisterHandlerFields const&) /-S/ydb/core/mon/mon.cpp:1543:9 (ydbd+0x1d3ed0fb) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #1 NKikimr::NViewer::TViewer::Bootstrap(NActors::TActorContext const&) /-S/ydb/core/viewer/viewer.cpp:216:26 (ydbd+0x33016bd9) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #2 NActors::TActorBootstrapped<NKikimr::NViewer::TViewer>::StateBootstrap(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor_bootstrapped.h:22:22 (ydbd+0x33015a2f) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #3 NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.cpp:406:17 (ydbd+0x1a1c1385) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #4 NActors::TExecutorThread::Execute(NActors::TMailbox*, bool) /-S/ydb/library/actors/core/executor_thread.cpp:268:28 (ydbd+0x1a2266c9) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #5 NActors::TExecutorThread::ProcessExecutorPool()::$_0::operator()(NActors::TMailbox*, bool) const /-S/ydb/library/actors/core/executor_thread.cpp:458:39 (ydbd+0x1a22ca1a) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #6 NActors::TExecutorThread::ProcessExecutorPool() /-S/ydb/library/actors/core/executor_thread.cpp:510:13 (ydbd+0x1a22c370) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #7 NActors::TExecutorThread::ThreadProc() /-S/ydb/library/actors/core/executor_thread.cpp:536:9 (ydbd+0x1a22d712) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #8 void* (anonymous namespace)::ThreadProcWrapper<ISimpleThread>(void*) /-S/util/system/thread.cpp:386:45 (ydbd+0x18c49702) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #9 (anonymous namespace)::TPosixThread::ThreadProxy(void*) /-S/util/system/thread.cpp:245:20 (ydbd+0x18c4cc5c) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   
E   Previous write of size 8 at 0x724400044cb8 by main thread (mutexes: write M0):
E   #0 NActors::TMon::Stop() /-S/ydb/core/mon/mon.cpp:1471:21 (ydbd+0x1d3ec1d8) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #1 NKikimr::TKikimrRunner::KikimrStop(bool) /-S/ydb/core/driver_lib/run/run.cpp:2008:21 (ydbd+0x2c1c3311) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #2 NKikimr::MainRun(NKikimr::TKikimrRunConfig const&, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:50:17 (ydbd+0x2a9b1b3a) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #3 NKikimr::NDriverClient::TClientCommandServer::Run(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/core/driver_lib/cli_utils/cli_cmds_server.cpp:50:12 (ydbd+0x2c0f050c) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #4 NYdb::NConsoleClient::TClientCommand::ValidateAndRun(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:315:16 (ydbd+0x2a9f68fa) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #5 NYdb::NConsoleClient::TClientCommandTree::Run(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:512:33 (ydbd+0x2a9f930b) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #6 NYdb::NConsoleClient::TClientCommand::ValidateAndRun(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:315:16 (ydbd+0x2a9f68fa) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #7 NYdb::NConsoleClient::TClientCommand::Process(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:257:16 (ydbd+0x2a9f5afd) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #8 NKikimr::NDriverClient::NewClient(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/cli_utils/cli_cmds_root.cpp:77:26 (ydbd+0x2a9ee8c0) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #9 NKikimr::Main(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:151:20 (ydbd+0x2a9b5855) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #10 ParameterizedMain(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:201:16 (ydbd+0x2a9b6ea3) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #11 main /-S/ydb/apps/ydbd/main.cpp:31:12 (ydbd+0x189dbe2e) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   
E   Location is heap block of size 320 at 0x724400044c00 allocated by main thread:
E   #0 operator new(unsigned long) /-S/contrib/libs/clang18-rt/lib/tsan/rtl/tsan_new_delete.cpp:64:3 (ydbd+0x18a80697) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #1 NKikimr::TKikimrRunner::InitializeMonitoring(NKikimr::TKikimrRunConfig const&, bool) /-S/ydb/core/driver_lib/run/run.cpp:496:26 (ydbd+0x2c1a895e) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #2 NKikimr::TKikimrRunner::CreateKikimrRunner(NKikimr::TKikimrRunConfig const&, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/run.cpp:2138:13 (ydbd+0x2c1c47a6) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #3 NKikimr::MainRun(NKikimr::TKikimrRunConfig const&, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:44:43 (ydbd+0x2a9b1a58) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #4 NKikimr::NDriverClient::TClientCommandServer::Run(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/core/driver_lib/cli_utils/cli_cmds_server.cpp:50:12 (ydbd+0x2c0f050c) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #5 NYdb::NConsoleClient::TClientCommand::ValidateAndRun(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:315:16 (ydbd+0x2a9f68fa) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #6 NYdb::NConsoleClient::TClientCommandTree::Run(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:512:33 (ydbd+0x2a9f930b) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #7 NYdb::NConsoleClient::TClientCommand::ValidateAndRun(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:315:16 (ydbd+0x2a9f68fa) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #8 NYdb::NConsoleClient::TClientCommand::Process(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:257:16 (ydbd+0x2a9f5afd) (BuildId: ea2b3091d619af82130a284fffe675e57f287956)
E   #9 NKikimr::NDriverClient::NewClient(int, char**, ...
```

--- a/ydb/core/mon/mon.cpp
+++ b/ydb/core/mon/mon.cpp
@@ -1537,10 +1537,10 @@ NMonitoring::IMonPage* TMon::RegisterCountersPage(const TString& path, const TSt
 }
 
 void TMon::RegisterActorHandler(const TRegisterHandlerFields& fields) {
+    TGuard<TMutex> g(Mutex);
     if (ActorSystem) {
         ActorSystem->Send(HttpMonServiceActorId, new TEvMon::TEvRegisterHandler(fields));
     } else {
-        TGuard<TMutex> g(Mutex);
         ActorMonPages.emplace_back(TActorMonPageInfo{
             .Handler = fields,
         });

